src/openrct2/ride/RideRatings.cpp

```diff
@@ -816,11 +870,181 @@ static void ride_ratings_score_close_proximity(RideRatingUpdateState& state, Til
 
 static void RideRatingsCalculate(RideRatingUpdateState& state, Ride& ride)
 {
-    auto calcFunc = RideRatingsGetCalculateFunc(ride.type);
-    if (calcFunc != nullptr)
+    RideRatingsDescriptor rrd = ride.GetRideTypeDescriptor().RatingsData;
```

Comment:
```suggestion
    const auto& rrd = ride.GetRideTypeDescriptor().RatingsData;
```
Probably best to avoid copying this and it should be const.